### PR TITLE
add option to fix data used for psd to segments

### PR DESCRIPTION
**Pull request by ahnitz**
I needed a way to ensure that the psd segments used *exactly* the same data as the strain data. This is how it used to work, but we moved away from that. To add that functionality back in I added a new option that is an alternative to the '--psd-num-segemnts' option which just fixes the times used to the segments.

### CODE REVIEW

File: pycbc/psd/__init__.py [hunk 1]
```diff
@@ -402,6 +413,8 @@ def generate_overlapping_psds(opt, gwstrain, flen, delta_f, flow,
         psd_segment_length, psd_segment_stride, psd_inverse_length, psd_output).
     gwstrain : Strain object
         The timeseries of raw data on which to estimate PSDs.
+    fd_segments: list, Optional
```
Comment (spxiwh) [inline pycbc/psd/__init__.py:416]: If this is optional, why is it not a key-word argument?

**Review by spxiwh — APPROVED** [review]
I think this all looks fine. There is a small comment, but I also notice that the test suite failed, which should be rectified.

**Comment by ahnitz** [comment]
@spxiwh I found an edge case where this didn't necessarily do what I want (i.e. if you also enable zero padding), so I'm putting as work in progress before merging.

**Comment by ahnitz** [comment]
Need to rethink how to do this

**Comment by spxiwh** [comment]
I guess the original solution to this (used in things like pycbc_single_template) was just to dump the data blocks used to an XML file, and then ensure that the same data ranges are read in to the PSD estimation code. This might mean that you read in quite a bit more data than you need though ... Maybe pycbc_estimate_psd could dump the PSD blocks that it uses. It feels like this would always be safer if you just give a list of the times uses to estimate PSDs in some way.
